invoices/invoiceregistry.go

```diff
@@ -151,32 +151,48 @@ func NewRegistry(cdb *channeldb.DB, expiryWatcher *InvoiceExpiryWatcher,
 // to the invoice expirt watcher.
 func (i *InvoiceRegistry) scanInvoicesOnStart() error {
 	var (
-		pending  map[lntypes.Hash]*channeldb.Invoice
-		currScan int
+		pending   map[lntypes.Hash]*channeldb.Invoice
+		removable []channeldb.InvoiceDeleteRef
+		currScan  int
 	)
 
 	scanFunc := func(scanCount int,
 		paymentHash lntypes.Hash, invoice *channeldb.Invoice) error {
 		if currScan != scanCount {
 			pending = make(map[lntypes.Hash]*channeldb.Invoice)
+			removable = make([]channeldb.InvoiceDeleteRef, 0)
 			currScan = scanCount
 		}
 
 		if invoice.IsPending() {
 			pending[paymentHash] = invoice
+		} else {
+			ref := channeldb.InvoiceDeleteRef{
+				PayHash:  paymentHash,
+				AddIndex: invoice.AddIndex,
+			}
+
+			if invoice.Terms.PaymentAddr != channeldb.BlankPayAddr {
+				ref.PayAddr = &invoice.Terms.PaymentAddr
+			}
+
+			removable = append(removable, ref)
```

Comment:
We should only mark the invoice for removal if it's _expired_. Existing settled invoices are still useful for record keeping purposes.

Reply:
I wasn't sure if there are users who may wish to remove settled invoice data too. (Perhaps if their DB grew large due to the number of settled invoices rather than the canceled/expired ones.)
Changed to only include canceled invoices. PTAL